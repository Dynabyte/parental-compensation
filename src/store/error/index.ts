export * from "./types";
export * from "./action";

export { default as ErrorReducer } from "./reducer";
