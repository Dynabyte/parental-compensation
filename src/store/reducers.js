import {
    combineReducers
} from 'redux';
import CardReducer from '../card/cardreducer'

export default combineReducers({
    cards: CardReducer
});