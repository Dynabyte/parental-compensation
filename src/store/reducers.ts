import {
    combineReducers
} from 'redux';
import CardReducer from '../features/card/CardReducer';

export default combineReducers({
    cards: CardReducer
});