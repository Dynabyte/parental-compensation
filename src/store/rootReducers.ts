import { combineReducers } from "redux";
import { ErrorReducer } from "./middlewares/error";
import {
  CardReducer,
  CriteriaReducer,
  SalaryInfoReducer,
  PriceBaseReducer
} from "../features";

export default combineReducers({
  critera: CriteriaReducer,
  priceBase: PriceBaseReducer,
  salaryInfo: SalaryInfoReducer,
  cards: CardReducer,
  errors: ErrorReducer
});
