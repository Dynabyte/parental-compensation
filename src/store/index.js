import {
    createStore,
    applyMiddleware
} from 'redux';
import {
    composeWithDevTools
} from 'redux-devtools-extension/developmentOnly';
import thunk from 'redux-thunk';

import rootReducers from './reducers'


const middleware = [thunk];

export const store = createStore(rootReducers, composeWithDevTools(applyMiddleware(...middleware)));