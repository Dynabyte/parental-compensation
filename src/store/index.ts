import { createStore, applyMiddleware, Store } from "redux";
import { composeWithDevTools } from "redux-devtools-extension/developmentOnly";
import thunk, {
  ThunkMiddleware,
  ThunkAction,
  ThunkDispatch
} from "redux-thunk";

import rootReducers from "./rootReducers";
import {
  SalaryInfoActionTypes,
  CriteriaActionTypes,
  PriceBaseActionTypes,
  CardActionTypes
} from "../features";

const middlewares = [thunk as ThunkMiddleware<AppState, AppActions>];
const middleWareEnhancer = applyMiddleware(...middlewares);

export type ThunkActionType = ThunkAction<void, AppState, null, AppActions>;
export type ThunkDispatchType = ThunkDispatch<AppState, any, AppActions>;

export type AppState = ReturnType<typeof rootReducers>;

export type AppActions =
  | CriteriaActionTypes
  | PriceBaseActionTypes
  | SalaryInfoActionTypes
  | CardActionTypes;

export const store: Store<AppState> = createStore(
  rootReducers,
  composeWithDevTools(middleWareEnhancer)
);
