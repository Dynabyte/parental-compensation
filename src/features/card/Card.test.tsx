import { CardComponent } from "./components/Card.component";
import CardrReducer from "./Card.reducer";
import {
  LOAD_CARD,
  LOAD_CARD_SUCCESS,
  LOAD_CARD_FAILURE,
  ICard
} from "./Card.types";
import { setCardsData } from "../../store/selector";
import { AppState } from "../../store";
import { AppActions } from "../../store";

describe("Testin Card Component", () => {
  let wrapper: any;
  let props: any;
  beforeEach(() => {
    wrapper = render(<CardComponent {...props} />);
  });

  it("Should render empty div if cards array is empty", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Testin CardReducer", () => {
  it("should setup default card values", () => {
    const state = CardrReducer(undefined, { type: "@@INIT" });

    expect(state).toEqual({
      loaded: true,
      data: []
    });
  });

  it("should load cards successfully", () => {
    const importantValuesToRenderCards: AppState = {
      priceBase: {
        pbb1: 47300
      },
      salaryInfo: {
        basicSalary: "36150",
        salaryModel: "RÃ¶rlig"
      }
    };
    const data: ICard[] = setCardsData(importantValuesToRenderCards);
    const action: AppActions = { type: LOAD_CARD_SUCCESS, data };

    const state = CardrReducer(undefined, action);

    expect(state).toEqual({
      loaded: true,
      data: [
        {
          ConvertedBasicSalary: 44645.25,
          max10PBB: 473000,
          max15PBB: 709500,
          excessFixedSalary: 62743,
          parentalSalaryUpto10PBB: 3887.67,
          parentalSalaryAbove10PBB: 4641.26,
          total: 8528.93
        }
      ]
    });

    expect(state.data).toHaveLength(1);
  });
});
