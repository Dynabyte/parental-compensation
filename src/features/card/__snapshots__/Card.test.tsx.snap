// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testin Card Component Should render empty div if cards array is empty 1`] = `
<styled.div
  size={5}
/>
`;
